src/runtime/routing/extends/i18n.ts

```diff
@@ -119,6 +125,20 @@ export function extendI18n<Context = unknown, TI18n extends I18n = I18n>(
 
     const globalComposer = getComposer(i18n)
 
+    if (experimental.autoImportTranslationFunctions) {
+      globalThis.$t = function (...args: unknown[]) {
```

Comment:
I think memory leaks will be occure with exporting to `globalThis`.

Especially on the server side, `setup` of `defineNuxtPlugin` is executed on every request. This means that `extendI18n` will be executed multiple times.

we need to release those functions assigned to `globalThis`.

A good way to relase them is to use `onScopeDispose`. 
https://vuejs.org/api/reactivity-advanced.html#onscopedispose

`extendI18n` creates a scope with `effectScope`. You can detach the scope from `globalThis` when it is destroyed, so we need to check or verify when `onScopeDispose` is called on the server side.

Reply:
Ah yes, I had a feeling this could cause leaks.

I'm also worried people will use these translation functions outside of setup in unexpected places which may cause further leaks or unexpected behavior.